File: 0189-Rotate_Array/0189-Rotate_Array.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var rotate = function(nums, k) {
    k = k % nums.length;
  
    let l = 0;
    let r = nums.length - 1;
    while (l < r) {
        [nums[l], nums[r]] = [nums[r], nums[l]]
        l += 1
        r -= 1
    }
    l = 0
    r = k -1
    while (l < r) {
        [nums[l], nums[r]] = [nums[r], nums[l]]
        l += 1
        r -= 1
    }
    l = k
    r = nums.length-1
    while (l < r) {
        [nums[l], nums[r]] = [nums[r], nums[l]]
        l += 1
        r -= 1
    }

};

File: 0189-Rotate_Array/0189-Rotate_Array.py
class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        k = k % len(nums)
        i = 0
        j = len(nums) -1
        while i < j:
            nums[i],nums[j] = nums[j], nums[i]
            i += 1
            j -= 1
        
        i = 0
        j = k -1
        while i < j:
            nums[i],nums[j] = nums[j], nums[i]
            i += 1
            j -= 1

        i = k
        j = len(nums) -1
        while i < j:
            nums[i],nums[j] = nums[j], nums[i]
            i += 1
            j -= 1
